windows/power_operation_scripts/set_power_script.yml

```diff
@@ -0,0 +1,152 @@
+# Copyright 2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+#   Set and check soft power operation
+# Parameters:
+#   power_script_op: power script operation
+#   power_script_type: default or custom
+
+- name: "Start to test {{ power_script_type }} {{ power_script_op }} script"
+  ansible.builtin.debug:
+    msg:
+      - "Start to test {{ power_script_type }} {{ power_script_op }} script"
+
+- name: "Set {{ power_script_type }} {{ power_script_op }} script"
+  include: set_power_script_{{ power_script_type }}.yml
+  vars:
+    power_cmd_op: "{{ power_script_op }}"
+
+- name: "Check the current configuration for {{ power_script_op }} script"
+  ansible.windows.win_shell: "& '{{ vmware_toolbox_cmd_path }}' script {{ power_script_op }} current"
+  register: power_script_result
+  delegate_to: "{{ vm_guest_ip }}"
+
+- name: "Print the current configuration for {{ power_script_op }} script"
+  ansible.builtin.debug: var=power_script_result
+
+- name: "Check current configuration is expected"
+  ansible.builtin.assert:
+    that:
+      - script_file_path
+      - power_script_result.stdout is defined
+      - power_script_result.stdout.rstrip() == script_file_path
+    fail_msg: "The {{ power_script_op }} script is {{ power_script_result.stdout | default('') }}, not expected {{ script_file_path }}."
+    success_msg: "The {{ power_script_op }} script is set to the expected one {{ power_script_result.stdout | default('') }}."
+
+- name: "Get content of file tools.conf"
+  ansible.windows.win_shell: "& type '{{ vmtools_config_file }}'"
```

Comment:
You can check all the places used "win_shell" can be changed to use `win_execute_cmd.yml` utils task except "retry" ones.